zookeeper-server/src/main/java/org/apache/zookeeper/common/X509Util.java

```diff
@@ -90,18 +121,30 @@ private static String[] getCBCCiphers() {
         return new String[]{"TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256", "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256", "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA", "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA", "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384", "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384", "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA", "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA"};
     }
 
-    private static String[] concatArrays(String[] left, String[] right) {
-        String[] result = new String[left.length + right.length];
-        System.arraycopy(left, 0, result, 0, left.length);
-        System.arraycopy(right, 0, result, left.length, right.length);
-        return result;
+    /**
+     * Returns a filtered set of ciphers, where ciphers not supported by the JDK are removed.
+     */
+    private static String[] getSupportedCiphers(String[]... cipherLists) {
+        Set<String> supported = new HashSet<>(Arrays.asList(
+            ((SSLServerSocketFactory) SSLServerSocketFactory.getDefault()).getSupportedCipherSuites()));
+
+        List<String> chosen = new ArrayList<>();
+        for (String[] ciphers : cipherLists) {
+            for (String cipher : ciphers) {
+                if (supported.contains(cipher)) {
+                    chosen.add(cipher);
+                }
+            }
+        }
+        return chosen.toArray(new String[0]);
```

Comment:
This is a pretty simple Stream operation that concatenates the lists, filters on whether the elements exist in the supported list, and then collects them into the list, all in one line (2 when wrapped):

```suggestion
        return Arrays.stream(cipherLists).flatMap(Arrays::stream).filter(supported::contains)
            .collect(Collectors.toList()).toArray(new String[0]);
```

Reply:
Changed according to suggestion.